presto-main/src/main/java/com/facebook/presto/operator/scalar/MapSubscriptOperator.java

```diff
@@ -97,6 +97,11 @@ private static MethodHandle lookupMethod(Type keyType, Type valueType)
         }
     }
 
+    public static Slice longvoidSubscript(Slice map, long key)
```

Comment:
I think there will be the same problem for the other key types (Slice, double...etc). Can you fix those too?